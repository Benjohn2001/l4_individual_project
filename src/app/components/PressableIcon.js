import React from "react";
import PropTypes from "prop-types";
import { Pressable, View } from "react-native";
import { Feather } from "@expo/vector-icons";

function PressableIcon(props) {
  const { onPress, icon, size, color } = props;
  return (
    <View className="flex-row">
      <Pressable testID="pressableIcon" onPress={onPress}>
        <Feather name={icon} size={size} color={color} />
      </Pressable>
    </View>
  );
}

PressableIcon.propTypes = {
  onPress: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
};

export default PressableIcon;
