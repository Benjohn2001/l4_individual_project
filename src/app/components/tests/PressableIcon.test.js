import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";
import PressableIcon from '../PressableIcon';


test('Check PressableIcon renders correctly and onpress works',()=>{
    const onPressMock = jest.fn();
    render(
        <PressableIcon 
            onPress={()=>{
                onPressMock()
            }}
            icon="users"
            size={20}
            color="black"
        />,
      );
      fireEvent.press(screen.getByTestId("pressableIcon"));
      expect(onPressMock)
      const domTree = renderer.create(
        <PressableIcon 
            onPress={()=>{
                onPressMock()
            }}
            icon="users"
            size={20}
            color="black"
        />
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})