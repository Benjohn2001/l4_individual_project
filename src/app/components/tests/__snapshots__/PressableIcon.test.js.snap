// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check PressableIcon renders correctly and onpress works 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableIcon"
  >
    <Text />
  </View>
</View>
`;
